website/server/controllers/api-v3/groups.js

```diff
@@ -759,19 +755,12 @@ api.leaveGroup = {
     }
 
     await group.leave(user, req.query.keep, req.body.keepChallenges);
-    if (group.hasNotCancelled()) await group.updateGroupPlan(true);
-    _removeMessagesFromMember(user, group._id);
-    await user.save();
 
-    if (group.type !== 'party') {
-      let guildIndex = user.guilds.indexOf(group._id);
-      user.guilds.splice(guildIndex, 1);
-    }
+    if (group.hasNotCancelled())  await group.updateGroupPlan(true);
 
-    let isMemberOfGroupPlan = await user.isMemberOfGroupPlan();
-    if (!isMemberOfGroupPlan) {
-      await payments.cancelGroupSubscriptionForUser(user, group);
-    }
+    _removeMessagesFromMember(user, group._id);
+
```

Comment:
Can you add the membercheck. It is out of scope of this PR. It is possible the logic is duplicate, but the check is different then below, so we should have test coverage and check for duplication separately.